File: csharp+vs+nunit/CarPlant/CarPlant.Tests/ToyotaCorollaTest.cs
using NUnit.Framework;
using System;

namespace CarPlant
{
	[TestFixture]
	public class ToyotaCorollaTest
	{

		private Car corolla;

		[SetUp]
		public void Init()
		{
			CarPlant.LOG = "";
			corolla = CarPlant.makeToyotaCorolla();
		}

		[Test]
		public void HasTheCorrectName()
		{
			Assert.AreEqual("Toyota Corolla", corolla.Name());
		}

		[Test]
		public void TurningLeftOnlyTurnsTheFrontWheels()
		{
			int degrees = TestHelpers.randomDegrees();
			corolla.TurnLeft(degrees);
			Assert.AreEqual("front left turning left " + degrees + " degrees\n"
			              + "front right turning left " + degrees + " degrees\n", CarPlant.LOG);
		}

		[Test]
		public void TurningRightOnlyTurnsTheFrontWheels()
		{
			int degrees = TestHelpers.randomDegrees();
			corolla.TurnRight(degrees);
			Assert.AreEqual("front left turning right " + degrees + " degrees\n"
			              + "front right turning right " + degrees + " degrees\n", CarPlant.LOG);
		}

		[Test]
		public void AcceleratingOnlyTurnsTheFrontWheels()
		{
			int speed = TestHelpers.randomSpeed();
			corolla.Accelerate(speed);
			Assert.AreEqual("front left accelerating " + speed + " kph\n"
			              + "front right accelerating " + speed + " kph\n", CarPlant.LOG);
		}

	}

}

File: java/src/com/xpsurgery/carplanttests/ToyotaCorollaTest.java
package com.xpsurgery.carplanttests;

import static org.junit.Assert.assertEquals;
import org.junit.Before;
import org.junit.Test;
import java.io.StringWriter;
import com.xpsurgery.carplant.CarPlant;
import com.xpsurgery.carplant.Car;

public class ToyotaCorollaTest {

	private Car corolla;
  private StringWriter log;

	@Before
	public void setUp() throws Exception {
    log = new StringWriter();
		corolla = CarPlant.makeToyotaCorolla(log);
	}

	@Test
	public void hasTheCorrectName() {
		assertEquals("Toyota Corolla", corolla.name());
	}

	@Test
	public void turningLeftOnlyTurnsTheFrontWheels() {
		int degrees = TestHelpers.randomDegrees();
		corolla.turnLeft(degrees);
		assertEquals("front left turning left " + degrees + " degrees\n"
				   + "front right turning left " + degrees + " degrees\n", log.toString());
	}

	@Test
	public void turningRightOnlyTurnsTheFrontWheels() {
		int degrees = TestHelpers.randomDegrees();
		corolla.turnRight(degrees);
		assertEquals("front left turning right " + degrees + " degrees\n"
				   + "front right turning right " + degrees + " degrees\n", log.toString());
	}

	@Test
	public void acceleratingOnlyTurnsTheFrontWheels() {
		int speed = TestHelpers.randomSpeed();
		corolla.accelerate(speed);
		assertEquals("front left accelerating " + speed + " kph\n"
				   + "front right accelerating " + speed + " kph\n", log.toString());
	}

}

